src/pages/audio-rewards-page/store/sagas.ts

```diff
@@ -179,6 +180,10 @@ function* claimChallengeRewardAsync(
               setVisibility({ modal: COGNITO_MODAL_NAME, visible: true })
             )
             break
+          case FailureReason.ALREADY_DISBURSED:
+            yield put(claimChallengeRewardAlreadyClaimed())
+            // Return out and do not retry. We've already earned this reward.
+            return
```

Comment:
Instead of returning here, maybe move `yield put(claimChallengeRewardWaitForRetry(claim))` into the cognito/hcaptcha cases?

Reply:
sgtm!